### ISSUE TITLE
[stable/postgresql] running charts with persistence.enabled=true with volumePermissions.securityContext.runAsUser=1001 causes permission errors

### ISSUE THREAD

**Issue by kpirateone** [helm/charts#18740]
**Describe the bug**
running the postgresql chart with volumePermissions.securityContext.runAsUser=1001.

receiving errors in the init container 'init-chmod-data':

```
chown: changing ownership of '/bitnami/postgresql/lost+found': Operation not permitted
chown: changing ownership of '/bitnami/postgresql': Operation not permitted
```

we cannot run this as root user in our environment 

this causes the primary container to not start.

if you choose to not run the 'init-chmod-data' (with volumePermissions.enabled=false) or if you modify the 'init-chmod-data' command script to beleive that it was successful (by appending any successful command like `echo foo` at the end)

the postgresql container fails to start up due to permission denied issues

```
│ postgresql 20:22:47.56 INFO  ==> ** Starting PostgreSQL setup **
│ postgresql 20:22:47.68 INFO  ==> Validating settings in POSTGRESQL_* env vars..
│ postgresql 20:22:47.69 INFO  ==> Loading custom pre-init scripts...
│ postgresql 20:22:47.70 INFO  ==> Initializing PostgreSQL database...
│ postgresql 20:22:47.72 INFO  ==> postgresql.conf file not detected. Generating it...
│ cp: cannot create regular file '/opt/bitnami/postgresql/conf/postgresql.conf': Permission denied
| postgresql 20:29:36.82 INFO  ==> Stopping PostgreSQL...
```

**Version of Helm and Kubernetes**:
helm: v2.13.1
kubernetes: v1.15.3

**Which chart**:
stable/postgresql: 6.5.5

**What happened**:
various permission errors when running init-container and postgresql

**What you expected to happen**:
i expected postgresql to complete setup necessary to start PostgreSQL

**How to reproduce it** (as minimally and precisely as possible):
run `helm template` with the following parameters:
 volumePermissions.securityContext.runAsUser=1001

apply generated deployment manifests

view errors (described above) in pod logs

**Anything else we need to know**:
please let me know if i can provide additional information

**Comment by juan131** [comment]
Hi @kpirateone 

Please correct me if I understood incorrectly but you are running this chart on a platform with restrictions that don't allow you to use the **initContainer** to adapt volume permissions, aren't you?

Did you try to disable the **initContainer** while keeping the securityContext enabled? I mean to do the equivalent to the command below:

```bash
$ helm install stable/postgresql --name my-release \
    --set volumePermissions.enabled=false \
    --set securityContext=true
```

Does your K8s distribution allow you to apply **securityContext** on your pods? I mean, do you have any [Pod Security Policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy) blocking them?

**Comment by kpirateone** [comment]
> Hi @kpirateone
> 
> Please correct me if I understood incorrectly but you are running this chart on a platform with restrictions that don't allow you to use the **initContainer** to adapt volume permissions, aren't you?
> 

yes that is correct, at least in the sense that i cannot run the init-chmod-data init-container as root (default: volumePermissions.securityContext.runAsUser=0) so i need to use another user, so i chose 1001 (which, fwiw, also doesnt look like the user exists on the minideb image)

https://github.com/helm/charts/tree/master/stable/postgresql

> Did you try to disable the **initContainer** while keeping the securityContext enabled? I mean to do the equivalent to the command below:
> 
> ```shell
> $ helm install stable/postgresql --name my-release \
>     --set volumePermissions.enabled=false \
>     --set securityContext=true
> ```
> 

yes, i do mention i tried `--set volumePermissions.enabled=false` and securityContext is enabled by default.  so, yes, this is where i get the other error regarding permissions as mentioned aboge

> Does your K8s distribution allow you to apply **securityContext** on your pods? I mean, do you have any [Pod Security Policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy) blocking them?

we can use securityContext, we just cannot run as user 0 or group 0

**Comment by juan131** [comment]
Hi @kpirateone 

> yes, i do mention i tried --set volumePermissions.enabled=false and securityContext is enabled by default. so, yes, this is where i get the other error regarding permissions as mentioned aboge
> we can use securityContext, we just cannot run as user 0 or group 0

Then sth must be wrong in your K8s setup since the **securityContext** should adapt the filesystem permissions of the PVC(s) that are requested for PostgreSQL.

Could you check whether the volumes permissions are correct (using a different chart/manifest) when a securityContext is specified?

**Comment by waterscar** [comment]
We got the same issue in our cluster (established by OpenStack and using Cinder as volume binding provider), kubernetes volume binding works differently among different providers and even different versions. I think the reason for having a `volumePermissions.enabled=true` and a initContainer was to handle these differences. But the one included in the chart can only run with root since it has `chown` command.

In our case, we disabled this and implemented our own initContainer, which basically does every step in the script exclude the `chmod` command.
